File: Linked Lists/Q13. Swapping Nodes in Linked List/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapNodes(ListNode head, int k) {
        ListNode a = head, c = head;
        int b = 1;
        while (b < k) {
            a = a.next;
            b++;
        }
        ListNode d = a;
        while (a.next != null) {
            a = a.next;
            c = c.next;
        }
        b = d.val;
        d.val = c.val;
        c.val = b;
        return head;
    }
}

File: Linked Lists/Q13. Swapping Nodes in Linked List/solution.py
class Solution(object):
    def swapNodes(self, head, k):
        """
        :type head: Optional[ListNode]
        :type k: int
        :rtype: Optional[ListNode]
        """
        first = head
        for _ in range(k - 1):
            first = first.next

        kth_from_start = first

        second = head
        while first.next:
            first = first.next
            second = second.next

        kth_from_end = second

        # swap values
        kth_from_start.val, kth_from_end.val = kth_from_end.val, kth_from_start.val

        return head
